draft-ietf-lamps-dilithium-certificates.md

```diff
@@ -577,6 +492,52 @@ key serialization. It for this reason the public key structure
 defined in {{ML-DSA-PubblicKey}} is intentionally encoded as a
 single OCTET STRING.
 
+## Rationale for disallowing HashML-DSA {#sec-disallow-hash}
+
+The HashML-DSA mode defined in Section 5.4 of {{FIPS204}} MUST NOT be
+used by CAs generating certificates or CRLs, CAs and RAs enrolling
+Subcribers, OCSP responders responding; in other words, public keys
+identified by `id-hash-ml-dsa-44-with-sha512`,
+`id-hash-ml-dsa-65-with-sha512`, and `id-hash-ml-dsa-87-with-sha512`
+MUST NOT be used in X.509 certificates and CRLs and related PKIX
+protocols. The notable exception is the public key in end-entity
+X.509 certificates; such public keys could be used beyond PKIX.
+
+This restriction is for both security and implementation reasons.
+
+The security reason for disallowing HashML-DSA is that the design of the
+ML-DSA algorithm provides enhanced resistance against signature
+collision attacks, compared with conventional RSA or ECDSA signature
+algorithms. Specifically, ML-DSA binds the hash of the public key `tr`
+to the message to-be-signed prior to hashing, as described in line 6 of
+Algorithm 7 of {{FIPS204}}. In practice, this provides binding to the
+indended verification public key, preventing some attacks that would
+otherwise allow a signature to be successfully verified against a
+non-intended public key. Also, this binding means that in the unlikely
+discovery of a collision attack against SHA-3, an attacker would
+have to perform a public-key-specific collision search in order to find
+message pairs such that `H(tr || m1) = H(tr || m2)` since a direct hash
+collision `H(m1) = H(m2)` will not suffice. HashML-DSA removes both of
+these enhanced security properties and therefore is a weaker signature
+algorithm.
```

Comment:
```suggestion
non-intended public key. Also, this unlikely, theoretical binding means that in the unlikely
discovery of a collision attack against SHA-3, an attacker would
have to perform a public-key-specific collision search in order to find
message pairs such that `H(tr || m1) = H(tr || m2)` since a direct hash
collision `H(m1) = H(m2)` will not suffice. HashML-DSA removes both of
these enhanced security properties.
```

Reply:
Sorry, more rephrase 

> Also, this binding means that in the unlikely, theoretical discovery of a collision attack against SHA-3